src/cdk-experimental/listbox/listbox.ts

```diff
@@ -64,15 +66,16 @@ export class CdkOption implements ListKeyManagerOption, Highlightable {
     this._disabled = coerceBooleanProperty(value);
   }
 
-  constructor(private _elementRef: ElementRef,
-              @Inject(forwardRef(() => CdkListbox)) public listbox: CdkListbox) {
+  constructor(private readonly _elementRef: ElementRef,
+              @Inject(forwardRef(() => CdkListbox)) readonly listbox: CdkListbox) {
   }
 
   /** Toggles the selected state, emits a change event through the injected listbox. */
   toggle() {
     if (!this._isInteractionDisabled()) {
       this.selected = !this.selected;
       this.listbox._emitChangeEvent(this);
+      this.listbox._updateSelectionModel(this);
```

Comment:
This is probably something we want to avoid, since now the option is mutating the listbox state. I would change this in one of two ways:
* Clicking the option emits on a stream, which the listbox listens to and updates its own state
* Handle clicks completely in the listbox instead of on the option